packages/clay-shared/src/Portal.tsx

```diff
@@ -10,24 +10,57 @@ const ClayPortalContext = React.createContext<React.RefObject<Element | null> |
 	null
 );
 
-ClayPortalContext.displayName = 'ClayPortalContext';
+const createElement = (nodeName: string, className?: string, id?: string) => {
+	const element = document.createElement(nodeName);
+
+	if (className) {
+		element.classList.add(className);
+	}
 
-export const ClayPortal: React.FunctionComponent<
-	React.HTMLAttributes<HTMLDivElement> & {
-		/**
-		 * Ref of element to render portal into.
-		 */
-		containerRef?: React.RefObject<Element>;
-
-		/**
-		 * Ref of element to render nested portals into.
-		 */
-		subPortalRef?: React.RefObject<Element>;
+	if (id) {
+		element.id = id;
 	}
-> = ({children, containerRef, subPortalRef}) => {
+
+	return element;
+};
+
+ClayPortalContext.displayName = 'ClayPortalContext';
+
+interface IProps extends React.HTMLAttributes<HTMLDivElement> {
```

Comment:
Hey @ambrinchaudhary, I think, it was not very clear, I'm sorry. The idea is that we can remove the `extend React.HTMLAttributes<HTMLDivElement>` from `IProps`, because we don’t need all the features of an `HTMLDivElement` and we don’t use it either, so in order not to provide the wrong types, we can just remove the `extend` and add the explicit type for `className` and `id`, just like you already did here. This way you can make `createElement` generic as before.

Reply:
Ok, I think I got you ;-). But I thinkg that removing the extend implies that `children` has to be typed as well....I'm getting errors on that but will work on this tomorrow. Thanks for your explanation.

Reply:
@ambrinchaudhary I forget that 😅. Yeah, this is also necessary, add the explicit type, it may have to be `React.ReactElement` and `React.ReactNode`, probably.

Reply:
Done in https://github.com/liferay/clay/pull/4067/commits/b9e8809ff1a06120aa0e417e1268b83c22315cf5. Although... I'm not sure if this is cleaner than passing directly the `className` and the `id` attributes, which are the ones that we  want to support... or do we really want to do this generic? Sure I'm missing something... so I let it up to you ;-) .

Reply:
@matuzalemsteles, just to add some context, @ambrinchaudhary and I were talking about the `createElement` function and we don't really understand why it needs to be generic. Could you tell us why you think this is needed?

Reply:
Of course. In my mind is that instead of manually defining `id` and `class` we can just isolate the `createElement` function from the rest, that is, it doesn’t know what attribute we are passing through or why we are just defining `id` and `class`, so when we see this at some point in the future it becomes simpler to look at it and be able to add new attributes if necessary.

So we were able to isolate `createElement` from the rest instead of attaching its signature to just `id` or `class`, for example, it is generic but we have full control of what we pass on to it, if we want to be more secure we can just `createElement('div', {id, class: className})`. Does it make sense?

Normally we may just want to define `id` and `class`, it also makes sense that this is not wrong. We also have the idea that generalizing too early can be bad but at some point in the future when we look at this, let's think about why we just defined that, why not simplify the method?, was that a use case?... I usually speak that when a person sees something in the code they will accept what they have and continue or they will refactor. In my view, this seems like a healthy change for us to keep around, even if it grows it seems that it's logic will not change.

Reply:
@matuzalemsteles I understand how this `createElement` function works, I just thought it might be unnecessary given that we're only going to create `div` elements and only set their `id` and `class` attributes. I'm not saying it's bad to support more names and attributes (and on top of that we're calling the function ourselves so there's no risk of seeing `createElement('foo', {bar: 'baz'})`) but if we really wanted to implement a "robust" `createElement` function it could quickly become complex. For example, are we  going to let pass any value for our `name` parameter, do we want to validate the attributes, etc... Since we're not exporting it, there's no harm done and we can push this forward as far as I'm concerned but I just wanted to check it with you.

Reply:
@julien yeah I agree, it's makes sense too. I think we can always try to find a balance, currently this implementation is not very complex, it allows to evolve, minimally generic to create an element with several native attributes, like `aria`... etc, so your responsibility is not to validate or deal with other extreme cases that we would have to deal with, like updating the values after they change, that would probably be somewhere else ... what I mean is that we can generalize a little bit just to not keep some things hardcoded, like manually defining the `element.id = id`. For example, as this is only used in this file which makes sense, we would have to do something like:

```js
const createDivElement = (
	className: string,
	id: string,
) => {
	const element = document.createElement('div');

	element.setAttribute('class', className);
	element.setAttribute('id', id);

	return element;
};
```

Does that make sense to you?

Reply:
Hey @matuzalemsteles I agree about finding a balance. Your suggestion is *exactly* what I had in mind, and my reasoning is the following: I'd rather not do anything more than what's needed to make it work. 

I'd go with the solution you suggested in the previous comment (just `className` and `id` parameters for now), and if we need to allow for more attributes in the future, we can always go with the more complete version (i.e [b7445bc](https://github.com/liferay/clay/commit/b7445bcd6f16cdda39ef7bd0af58bda2d3b1949e)) (after a second thought, attribute validation seems like a wrong idea, since even the native APIs let you create any kind of element with any attribute without throwing exceptions :smile: ) . 

I *have to admit* that the current change doesn't add a lot complexity, so you could say that I "complained for nothing"; but  it's always a good thing to take a moment and think about a change to be sure that we're all on the same track. Thanks!

Reply:
Ok then, I added a new commit with simplifying our function. I doubt about putting back or not the extend of `React.HTMLAttributes<HTMLDivElement>,` but if we are not using it, better not to add it, right?

Reply:
@julien don't worry, this is good for evaluating our options 🙂.

>  Ok then, I added a new commit with simplifying our function. I doubt about putting back or not the extend of React.HTMLAttributes<HTMLDivElement>, but if we are not using it, better not to add it, right?

Yes, if it is not being used then I think it is better not to add it. Because when we use the component we will be seeing a typing that is not true, accepting properties that are not actually being accepted.